packages/docusaurus-plugin-pwa/src/registerSw.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+async function clearRegistrations() {
+  const registrations = await navigator.serviceWorker.getRegistrations();
+  registrations.forEach((registration) => {
+    registration.unregister();
+  });
+
+  window.location.reload();
+}
+
+const MAX_MOBILE_WIDTH = 940;
+const APP_INSTALLED_KEY = 'docusaurus.pwa.appInstalled';
+
+(async () => {
+  if (typeof window === 'undefined') {
+    return;
+  }
+
+  if ('serviceWorker' in navigator) {
+    const {Workbox} = await import('workbox-window');
+
+    const shouldCacheFiles =
+      window.innerWidth <= MAX_MOBILE_WIDTH ||
+      (navigator.connection && navigator.connection.saveData) ||
+      localStorage.getItem(APP_INSTALLED_KEY) ||
+      window.location.search.includes('offline') ||
+      process.env.ALWAYS_PRECACHE === 'true';
+
+    const enabledParam = shouldCacheFiles ? `?enabled` : '';
+    const swUrl = `${process.env.SERVICE_WORKER}${enabledParam}`;
+    const wb = new Workbox(swUrl);
+    const registration = await wb.register();
+    const type = 'SKIP_WAITING';
+
+    const handleServiceWorkerWaiting = async () => {
+      // Immediately load new service worker when files aren't cached
+      if (!shouldCacheFiles) {
+        wb.messageSW({type});
+      } else if (process.env.PWA_POPUP) {
+        const renderPopup = (await import('./renderPopup')).default;
```

Comment:
Doc says that dynamic imports in SW is not advised.
https://developers.google.com/web/tools/workbox/guides/using-bundlers#code_splitting_and_dynamic_imports

But I think due to explicitly disabling code splitting, it works right?